UnityProject/Assets/Scripts/Objects/Engineering/SuperMatter.cs

```diff
@@ -1065,16 +1069,45 @@ private IEnumerable<Collider2D> GetNearbyEntities(Vector3 centrepoint, int mask,
 
 		private void AddMessageToChat(string message, bool sendToCommon = false)
 		{
-			Chat.AddCommMsgByMachineToChat(gameObject, message, ChatChannel.Engineering,
-				Loudness.SCREAMING,  broadcasterName: "Supermatter Warning System: ", language: LanguageManager.Common);
+			ChatEvent chatEvent = new ChatEvent();
+			chatEvent.message = message;
+			chatEvent.speaker = "Supermatter Warning System: ";
+			chatEvent.VoiceLevel = Loudness.SCREAMING;
+			chatEvent.position = registerTile.WorldPositionServer;
+			chatEvent.originator = gameObject;
+
+			chatEvent.channels = ChatChannel.Engineering;
+			
+			InfluenceChat(chatEvent);
 
 			if (sendToCommon)
 			{
-				Chat.AddCommMsgByMachineToChat(gameObject, message, ChatChannel.Common,
-					Loudness.SCREAMING, broadcasterName: "Supermatter Warning System: ", language: LanguageManager.Common);
+				chatEvent.channels = ChatChannel.Common;
```

Comment:
```suggestion
				chatEvent.channels |= ChatChannel.Common;
```

Bitwise OR so it goes to both common and engineering